### PR TITLE
[CBW-1440] added support for concordiumwallet://wc deeplink

### PR DESCRIPTION
**Pull request by TouHouNo**
## Changes

 added support for new concordiumwallet://wc deeplink

for testing please push an intent with the command `adb shell am start -W -a android.intent.action.VIEW -d "concordiumwallet://wc" software.concordium.mobilewallet.seedphrase.testnet`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -53,6 +53,16 @@
 
                 <data android:scheme="wc" />
             </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
```
Comment (orhoj) [inline app/src/main/AndroidManifest.xml:62]: You can just put this `<data>` clause inside the existing `intent-filter` above.

File: app/src/main/java/com/concordium/wallet/ui/MainActivity.kt [hunk 2]
```diff
@@ -61,7 +61,11 @@ class MainActivity : BaseActivity(), IdentityStatusDelegate by IdentityStatusDel
 
         initializeViews()
 
-        intent?.data?.let { if (it.toString().startsWith("wc")) wcUri = it.toString() }
+        intent?.data?.let {
+            if (it.toString().startsWith("wc") || it.toString()
```
Comment (orhoj) [inline app/src/main/java/com/concordium/wallet/ui/MainActivity.kt:65]: These lines are duplicated twice below, so create a helper function to perform the check instead.

I don't really understand why it needs to set `wcUri` in both places in the intents, but I guess it's easier to just keep that as is.

File: app/src/main/java/com/concordium/wallet/ui/recipient/scanqr/ScanQRViewModel.kt [hunk 3]
```diff
@@ -46,6 +46,7 @@ class ScanQRViewModel(application: Application) : AndroidViewModel(application)
     }
 
     fun checkQrWalletConnect(qrInfo: String): Boolean {
-        return qrInfo.isNotBlank() && qrInfo.lowercase().startsWith("wc:")
+        return qrInfo.isNotBlank() && (qrInfo.lowercase().startsWith("wc:") || qrInfo.lowercase()
```
Comment (orhoj) [inline app/src/main/java/com/concordium/wallet/ui/recipient/scanqr/ScanQRViewModel.kt:49]: For the QR code we don't need to support anything but `wc` for the format. The QR codes generated by WalletConnect are all just using `wc:`, so unless someone built something custom we won't see such QR codes. And I guess that won't be needed as you are already in the app when scanning.

**Review by orhoj — CHANGES_REQUESTED** [review]

**Comment by orhoj** [comment]
Will wait to approve until Shashi has tested.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**